modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -2697,6 +2707,14 @@ private RestoreLogicalState applyLogicalUpdates(
                         }
 
                         break;
+
+                    case PARTITION_CLEARING_STARTED:
+                        PartitionClearingStarted rec0 = (PartitionClearingStarted) rec;
```

Comment:
If don't miss something, it looks like we can start clearing the partition right here. Yes, this requires that all data updates from the WAL should be applied under the corresponding lock. Please take a look at `applyUpdates(..., boolean lockEntry)`

Reply:
It didn't work with `localPartition(rec0.partId())` but works with `forceCreatePartition(rec0.partId())`
I don't understand about `applyUpdates(..., boolean lockEntry)`

Reply:
Well, I was wrong. We cannot run clearing tasks in parallel with applying data records.
I propose the following solution:
```
stripedApply(() -> {
    try {
        part.clearVersion(rec0.clearVersion());

        IgniteInternalFuture<?> clearFut = grp.
            shared().
            evict().
            evictPartitionAsync(grp, part, new GridFutureAdapter<>());

        clearFut.get();

        part.updateClearVersion();
    }
    catch (IgniteCheckedException e) {
        U.error(log, "Failed to apply partition clearing record, " + rec0);

        applyError.compareAndSet(null, e);
    }
}, rec0.groupId(), rec0.partitionId(), exec, semaphore);
```